pkg/image/admission/admission.go

```diff
@@ -0,0 +1,319 @@
+	}
+
+	if len(images) == 0 {
+		glog.V(4).Infof("%s: unable to validate image limit ranges for %#v, no images were found", pluginName, obj)
+		return nil
+	}
+
+	errs := []error{}
+
+	for _, img := range images {
+		if value, exists := img.Annotations[imageapi.ManagedByOpenShiftAnnotation]; !exists || value != "true" {
+			glog.V(5).Infof("%s: image is not managed by openshift, ignoring", pluginName)
+			continue
+		}
+
+		// ensure metadata is loaded from the manifest, this may not be automatically filled in if
+		// an older registry is in use
+		err := imageapi.ImageWithMetadata(&img)
+		if err != nil {
+			errs = append(errs, err)
+			continue
+		}
+
+		for _, limit := range limitRange.Spec.Limits {
+			err := AdmitImage(img.DockerImageMetadata.Size, limit)
+			if err != nil {
+				errs = append(errs, err)
+			}
+		}
+	}
+
+	if len(errs) != 0 {
+		return utilerrors.NewAggregate(errs)
+	}
+
+	return nil
+}
+
+// getImageStreamImages gives back a slice of images that need checked against the limit range
+// based on the image stream.  It filters out preexisting images that do not need to be checked
+// by using the previousIS.
+func (a *imageLimitRangerPlugin) getImageStreamImages(is, previousIS *imageapi.ImageStream) (map[string]imageapi.Image, error) {
+	computer := imagequota.NewGenericImageStreamUsageComputer(a.osClient, true, false)
+
+	previousImages := map[string]imageapi.Image{}
+	if previousIS != nil {
+		// get images referenced in the previous image stream so we can filter them out of
+		// the new images stream so we are only guarding against new images
+		handler := newImageRecordingHandler(previousImages)
+		err := computer.ProcessImageStreamImages(previousIS, handler)
+		if err != nil {
+			return nil, err
+		}
+	}
+
+	images := map[string]imageapi.Image{}
+	handler := newImageRecordingHandler(images)
+	err := computer.ProcessImageStreamImages(is, handler)
+	if err != nil {
+		return nil, err
+	}
+
+	// filter out anything in previous
+	imagesToProcess := map[string]imageapi.Image{}
+	for _, img := range images {
+		if _, ok := previousImages[img.Name]; !ok {
+			imagesToProcess[img.Name] = img
+		}
+	}
+
+	glog.V(4).Infof("%s: prevImages: %v, newImages: %v, toProcess: %v", pluginName, len(previousImages), len(images), len(imagesToProcess))
+
+	return imagesToProcess, nil
+}
+
+// SetOpenshiftClient provides init functionality for anything that requires the OpenShift client.
+func (a *imageLimitRangerPlugin) SetOpenshiftClient(c client.Interface) {
+	a.osClient = c
+
+	reflector := cache.NewReflector(
+		&cache.ListWatch{
+			ListFunc: func(options kapi.ListOptions) (runtime.Object, error) {
+				return a.osClient.ImageStreams(kapi.NamespaceAll).List(options)
+			},
+			WatchFunc: func(options kapi.ListOptions) (watch.Interface, error) {
+				return a.osClient.ImageStreams(kapi.NamespaceAll).Watch(options)
+			},
+		},
+		&imageapi.ImageStream{},
+		a.imageStreamStore,
+		0,
+	)
+	a.imageStreamReflector = reflector
+
+	a.Run()
```

Comment:
There isn't really a good place to do this since there are limited lifecycle methods for admission plugins.  Mentioned here: https://github.com/openshift/origin/pull/7930#discussion_r56385652.  I'd also like to not have to set up reflectors in a setter method.  It's an unexpected side effect but it's the only way I can see to ensure the client is available.